### PR TITLE
Issue #2244 . Replace Programming guide PDF with a Programming Guide md file

### PR DESCRIPTION
**Pull request by jeffmiles63**
Issue #2244 

Kokkos-Programming-Guide.md contains a link to the kokkos/ProgrammingGuide wiki

### CODE REVIEW

**Comment by crtrott** [comment]
Can we squash the first three commits (or all) of them?

**Comment by jeffmiles63** [comment]
done

File: doc/Kokkos-Programming-Guide.md [hunk 1]
```diff
@@ -0,0 +1 @@
+[Programming Guide](https://github.com/kokkos/ProgrammingGuide/wiki)
```
Comment (crtrott) [inline doc/Kokkos-Programming-Guide.md]: Ups this should be kokkos/kokkos/wiki , ProgrammingGuide contains the private version.

Comment (jeffmiles63) [reply doc/Kokkos-Programming-Guide.md]: updated

**Pull request merged**